helm/dagger/templates/engine-daemonset.yaml

```diff
@@ -0,0 +1,103 @@
+{{- if .Values.magicache.enabled }}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "dagger.fullname" . }}-magicache-token
+  namespace:  {{ .Release.Namespace }}
+data:
+  _EXPERIMENTAL_DAGGER_CACHESERVICE_TOKEN: {{ required "A magicache token is required" .Values.magicache.token | b64enc | quote }}
+{{- end }}
+{{- if .Values.engine.config }}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "dagger.fullname" . }}-engine-config
+  namespace: {{ .Release.Namespace }}
+data:
+  engine.toml: |{{- .Values.engine.config | nindent 4 }}
+{{- end }}
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ include "dagger.fullname" . }}-engine
```

Comment:
add a checksum label for the Secret and Configmap so resource gets updated if those change in the future. i.e: 

```
spec:
  template:
    metadata:
      annotations:
        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
        checksum/secret: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
```

this requires placing the Secret and the ConfigMap on it's own files but it's generally a good practice.

Reply:
Yes! 💯